projects/plugins/boost/app/assets/src/js/features/page-cache/page-cache.tsx

```diff
@@ -1,34 +1,290 @@
+			</>
+		);
+	};
+
+	const loggingEnabledMessage = __( 'Logging enabled.', 'jetpack-boost' );
+	const loggingDisabledMessage = __( 'Logging disabled.', 'jetpack-boost' );
+	useMutationNotice( 'update-bypass-patterns', mutateBypassPatterns );
+	useMutationNotice( 'update-logging', mutateLogging, {
+		successMessage: logging ? loggingEnabledMessage : loggingDisabledMessage,
+	} );
+
+	useMutationNotice( 'clear-page-cache', runClearPageCacheAction, {
+		savingMessage: __( 'Clearing cache…', 'jetpack-boost' ),
+		errorMessage: __( 'Unable to clear cache.', 'jetpack-boost' ),
+		successMessage: clearedCacheMessage || __( 'Cache cleared.', 'jetpack-boost' ),
+	} );
+
+	return (
+		pageCache && (
+			<div className={ styles.wrapper }>
+				<div className={ styles.head }>
+					<div className={ styles.summary }>{ getSummary() }</div>
+					<div className={ styles.actions }>
+						<Button
+							variant="link"
+							size="small"
+							weight="regular"
+							iconSize={ 16 }
+							icon={ <Lightning /> }
+							onClick={ clearPageCache }
+							disabled={ runClearPageCacheAction.isPending }
+						>
+							{ __( 'Clear Cache', 'jetpack-boost' ) }
+						</Button>{ ' ' }
+						<Button
+							variant="link"
+							size="small"
+							weight="regular"
+							iconSize={ 16 }
+							icon={ isExpanded ? <ChevronUp /> : <ChevronDown /> }
+							onClick={ () => setIsExpanded( ! isExpanded ) }
+						>
+							{ __( 'Show Options', 'jetpack-boost' ) }
+						</Button>
+					</div>
+				</div>
+				{ isExpanded && (
+					<div className={ styles.body }>
+						<>
+							<BypassPatterns
+								patterns={ bypassPatterns.join( '\n' ) }
+								setPatterns={ patterns =>
+									mutateBypassPatterns.mutate( patterns.split( '\n' ).map( item => item.trim() ) )
+								}
+								showErrorNotice={ mutateBypassPatterns.isError }
+							/>
+							<div className={ styles.section }>
+								<div className={ styles.title }>{ __( 'Logging', 'jetpack-boost' ) }</div>
+								<label htmlFor="cache-logging" className={ styles[ 'logging-toggle' ] }>
+									<input
+										type="checkbox"
+										id="cache-logging"
+										checked={ logging }
+										onChange={ event => mutateLogging.mutate( event.target.checked ) }
+									/>{ ' ' }
+									{ __( 'Activate logging to track all your cache events.', 'jetpack-boost' ) }
+								</label>
+								{ logging && (
+									<Link className={ styles[ 'see-logs-link' ] } to="/cache-debug-log">
+										{ __( 'See Logs', 'jetpack-boost' ) }
+									</Link>
+								) }
+								<div className={ styles.clearfix } />
+							</div>
+						</>
+					</div>
+				) }
+			</div>
+		)
+	);
 };
 
-const PageCache = ( { setup, error }: Props ) => {
+type BypassPatternsProps = {
+	patterns: string;
+	setPatterns: ( newValue: string ) => void;
+	showErrorNotice: boolean;
+};
+
+const BypassPatterns = ( {
+	patterns,
+	setPatterns,
+	showErrorNotice = false,
+}: BypassPatternsProps ) => {
```

Comment:
This file seems to contain multiple meaty components. Would it make sense to split it up?

Can be a separate PR.